### ISSUE TITLE
AV1 Film grain handling and HDR10+

### ISSUE THREAD

**Issue by alexismt73** [AOMediaCodec/av1-hdr10plus#2]
One of the advanced features of the AV1 specification is film grain synthesis. Noise, such as film grain, could be removed at the source and prior to encoding, while information to resynthesize the noise at the decoder can be included in the AV1 bitstream. The AV1 specification requires that such noise is always added during reconstruction, but allows some flexibility on the specifics of the noise synthesis and the addition, to better cater for different architectures. For example, a decoder may decide to strictly follow the specification and in this case the output of the decoder will include the film grain, while another decoder may decide to only reconstruct the video signal up to and prior to the film grain synthesis. In this case, the film grain parameters can be passed to a secondary process, e.g. the display, where that process will determine in which domain/format the noise can be added. Different domains/formats include changes in chroma sampling, resolution, color representation (e.g. RGB), among others.

This raises the question of how HDR10+ should be handled when film grain is signalled in the AV1 bitstream. If the HDR10+ metadata are based on the original content (with the grain) then the process should account of where the grain is to be added in the decoding/display chain. Some clarification may be needed in this document of how this should be handled.

**Comment by quantizationbit** [comment]
HDR10+ would be calculated based on the original content with the grain. This way one metadata can be used with any video codec.

**Comment by alexismt73** [comment]
I think the issue here is about where the metadata should be applied and how. If the metadata is calculated with the noise, it is important then that the processing Is applied only after film grain synthesis and addition is performed. A note in that regard, suggesting that "it is recommended that HDR10+ processing is applied after any film grain noise synthesis and addition is performed or, alternatively, the film grain noise synthesis would need to be adjusted appropriately." I am not sure exactly how the later would work so maybe the alternative case could be skipped if there is some uncertainty of how that is done.

**Comment by quantizationbit** [comment]
If HDR10+ and codec are considered separable then doesn't that imply that it is the codec's job to deliver baseband video (including film grain) to downstream processes such as HDR10+?  Another way to look at it is if there is an HDMI input it is certainly separable and HDR10+ processing is on final picture.

**Comment by alexismt73** [comment]
Not in this case. The way the specification is written it allows film grain to be part of the display process, not part of the decoding process. So the output of the decoder is the content without the noise + the metadata for the noise generation.

**Comment by podborski** [comment]
Here is the proposed text for the **Film Grain Processing** section:

> It is possible that some AV1 coded bitstreams may contain both HDR10+ metadata and film grain synthesis information. It is strongly recommended that in such scenarios the film grain synthesis is performed prior to any HDR10+ processing.

Anything else we could add?
